pyro/distributions/lkj.py

```diff
@@ -0,0 +1,197 @@
+        z = x.tanh().clamp(-1 + eps, 1 - eps)
+        return _vector_to_l_cholesky(z)
+
+    def _inverse(self, y):
+        if (y.shape[-2] != y.shape[-1]):
+            raise ValueError("A matrix that isn't square can't be a Cholesky factor of a correlation matrix")
+        D = y.shape[-1]
+
+        z_tri = y.new_zeros(y.shape[:-2] + (D - 2, D - 2))
+        z_stack = [
+            y[..., 1:, 0]
+        ]
+
+        for i in range(2, D):
+            z_tri[..., i - 2, 0:(i-1)] = y[..., i, 1:i] / (1-y[...,i,0:(i-1)].pow(2).cumsum(-1)).sqrt()
+        for j in range(D - 2):
+            z_stack.append(z_tri[..., j:, j])
+
+        z = torch.cat(z_stack, -1)
+        return torch.log1p((2*z)/(1-z)) / 2
+
+    def log_abs_det_jacobian(self, x, y):
+        # Note dependence on pytorch 1.0.1 for batched tril
+        tanpart = x.cosh().log().sum(-1).mul(-2)
+        matpart = (1 - y.pow(2).cumsum(-1).tril(diagonal=-2)).log().div(2).sum(-1).sum(-1)
+        return tanpart + matpart
+
+# register transform to global store
+@biject_to.register(corr_cholesky_constraint)
+@transform_to.register(corr_cholesky_constraint)
+def _transform_to_corr_cholesky(constraint):
+    return CorrLCholeskyTransform()
+
+
+########################################
+# Define distribution
+########################################
+
+# TODO: Modify class to support more than one eta value at a time?
+class CorrLCholeskyLKJPrior(TorchDistribution):
+    """
+    Generates cholesky factors of correlation matrices using an LKJ prior.
+
+    The expected use is to combine it with a vector of variances and pass it
+    to the scale_tril parameter of a multivariate distribution such as MultivariateNormal.
+
+    E.g., if \theta is a (positive) vector of covariances with the same dimensionality
+    as this distribution, and \Omega is sampled from this distribution, scale_tril=torch.mm(torch.diag(sqrt(\theta)), \Omega)
+
+    Note that the `event_shape` of this distribution is `[d, d]`
+
+    :param int d: Dimensionality of the matrix
+    :param torch.Tensor eta: A single positive number parameterizing the distribution.
+    """
+    arg_constraints = {"eta": constraints.positive}
+    support = corr_cholesky_constraint
+    has_rsample = False
+
+    def __init__(self, d, eta):
+        if not torch.is_tensor(eta):
+            raise ValueError("Eta must be a tensor")
+        if any(eta <= 0):
+            raise ValueError("eta must be > 0")
+        vector_size = (d * (d - 1)) // 2
+        alpha = eta.add(0.5 * (d  - 1.0))
+
+        concentrations = eta.new_empty(vector_size,)
+        i = 0
+        for k in range(d-1):
+            alpha -= .5
+            for j in range(k+1, d):
+                concentrations[..., i] = alpha
+                i += 1
+        self._generating_distribution = Beta(concentrations, concentrations)
+        self._eta = eta
+        self._d = d
+        self._lkj_constant = None
+        self._event_shape = torch.Size((d, d))
```

Comment:
Call the parent constructor here, `super(CorrLCholeskyLKJPrior, self).__init__(batch_shape, event_shape, validate_args=validate_args)`.

Reply:
Thanks, I will address in the context of resolving the comment concerning `.expand`

Reply:
Should be resolved in what I just pushed.